controller/src/main/java/org/jboss/as/controller/descriptions/DefaultResourceDescriptionProvider.java

```diff
@@ -167,4 +172,17 @@ private ModelNode addDeprecatedInfo(final ModelNode model) {
         return deprecated;
     }
 
+    private PathAddress createAliasPathAddress(PathAddress pa) {
+        PathAddress aliasAddress = PathAddress.EMPTY_ADDRESS;
+        Iterator<PathElement> iter = pa.iterator();
+        while (iter.hasNext()) {
+            PathElement elt = iter.next();
+            if (elt.isWildcard()) {
+                aliasAddress = aliasAddress.append(PathElement.pathElement(elt.getKey(), "$" + elt.getKey()));
+            } else {
+                aliasAddress = aliasAddress.append(elt);
+            }
+        }
+        return aliasAddress;
+    }
```

Comment:
Please document this. It's quite clever. :) But the cleverness makes it confusing.